### DISCUSSION TITLE
Do you recommend webgl1 or webgl2 or is there no difference?

### DISCUSSION THREAD

**Discussion by goldenratio** [jnsmalm/pixi3d#123] [Q&A] [answered]
Hey,

sometimes I get warnings like this,
> WARNING: texture bound to texture unit 1 is not renderable. It might be non-power-of-2 or have incompatible texture filtering (maybe)?

Do you recommend webgl1 or webgl2 or is there no difference?

**Comment by jnsmalm** [comment, marked as answer]
Hi!

If WebGL2 is available I think it's better to use that. Almost all features in Pixi3D works with both versions, but you may encounter some bugs on older devices when running in WebGL1 when WebGL2 is available.

Do you have a reproducible scenario when you get that warning?

**Reply by goldenratio** [reply]
with webgl2, I get this error (can't share model unfortunately). Model does has have PIXI filter as post processing. Model suddenly disappears with below warnings.
Not sure, If this is helpful.

```
12800:14:06.542 [.WebGL-0x11416639c00] GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is incomplete: Attachments are not all the same size.
12800:14:06.543 [.WebGL-0x11416639c00] GL_INVALID_FRAMEBUFFER_OPERATION: Draw framebuffer is incomplete
00:14:08.562 localhost/:1 WebGL: too many errors, no more errors will be reported to the console for this context.
```

**Reply by jnsmalm** [reply]
Is it possible to reproduce error with another model you can share?
